jOOL/src/main/java/org/jooq/lambda/Agg.java

```diff
@@ -886,6 +970,126 @@ else if (percentile == 1.0)
+     * <p>This method will sort the elements after applying the <code>function</code> based on
+     * <code>comparator</code> at first, then it will find the all eligible results.
+     * @param <T>           The type of element
+     * @param <U>           The type of result of <code>function</code>
+     * @param percentile    The percentile number
+     * @param function      The given function
+     * @return              A {@link Seq} of type <code>T</code>
+     */
+    public static <T, U extends Comparable<? super U>> Collector<T, ?, Seq<T>> percentileAllBy(final double percentile, final Function<? super T, ? extends U> function) {
+        return percentileAllBy(percentile, function, naturalOrder());
+    }
+
+    /**
+     * Get a {@link Collector} that calculates the derived <code>PERCENTILE_DISC(percentile)</code> function
+     * given a specific ordering.
+     * <p>This method will sort the elements after applying the <code>function</code> based on
+     * <code>comparator</code> at first, then it will find the all eligible results.
+     * @param <T>           The type of element
+     * @param <U>           The type of result of <code>function</code>
+     * @param percentile    The percentile number
+     * @param function      The given function
+     * @param comparator    The given comparator
+     * @return              A {@link Seq} of type <code>T</code>
+     * @throws IllegalArgumentException when the percentile is less than 0 or more than 1
+     */
+    public static <T, U> Collector<T, ?, Seq<T>> percentileAllBy(final double percentile, final Function<? super T, ? extends U> function, final Comparator<? super U> comparator) {
+        if (percentile < 0.0 || percentile > 1.0) {
+            throw new IllegalArgumentException("Percentile must be between 0.0 and 1.0");
+        }
+
+        if (percentile == zero) {
+            return minAllBy(function, comparator);
+        } else if (percentile == one) {
+            return maxAllBy(function, comparator);
+        }
+
+        return Collector.of(
+                (Supplier<ArrayList<Tuple2<T, U>>>) ArrayList::new,
+                (l, v) -> l.add(tuple(v, function.apply(v))),
+                (l1, l2) -> {
+                    l1.addAll(l2);
+                    return l1;
+                },
+                l -> {
+                    final int size = l.size();
+
+                    if (size == zero) {
+                        return Seq.empty();
+                    } else if (size == one) {
+                        return Seq.of(l.get(0).v1);
+                    }
+
+                    l.sort(Comparator.comparing(t -> t.v2, comparator));
+                    // x.5 should be rounded down
+                    final int index = (int) -Math.round(-(size * percentile + half)) - 1;
+                    final T element = l.get(index).v1;
+                    return Seq.seq(l.stream().filter(t -> t.v1.equals(element)).map(t -> t.v1));
```

Comment:
It seems to me that in edge cases, this additional `O(N)` traversal is more expensive than a possibly more optimal way to find the "ties" of the result.

In addition to this, I wonder what the best way to implement `percentileAll()` would be. Is it really desireable to find all ties? Or only ties that are located *after* the percentile position?

This question is already answered partially by the behaviour of `maxAllBy()`, which produces *all* ties, regardless of their position with respect to the percentile value (1.0).

I'm undecided. And once more, I wonder whether we should add this functionality, or wait for someone who actually needs it to think about this more thoroughly.

I prefer the latter.